std/format.d

```diff
@@ -2066,7 +2231,75 @@ if (is(FloatingPointTypeOf!T) && !is(T == enum) && !hasToString!(T, Char))
 
     enforceFmt(n >= 0,
         "floating point formatting failure");
-    put(w, buf[0 .. min(n, buf.length-1)]);
+
+    auto len = min(n, buf.length-1);
+    ptrdiff_t dot = buf.indexOf('.');
+    if (fs.flSeparator && dot != -1)
+    {
+        ptrdiff_t firstDigit = buf.indexOfAny("0123456789");
+        ptrdiff_t ePos = buf.indexOf('e');
```

Comment:
Found the problem: `buf` is a static array of length 512, so the above searches of `buf` will actually search all 512 bytes, since in D, we do not stop upon encountering a null terminator. So lines 2236, 2239, and 2240 should use `buf[0 .. len].indexOf(...)` rather than `buf.indexOf(...)`.